tests/high-level-tests.lisp

```diff
@@ -64,3 +64,61 @@
     ;; Check that it yields teh right dimensions.
     (is (= (matrix-rows xxx) (matrix-cols xxx) (expt matrix-dim 3)))
     ))
+
+(deftest test-svd ()
+  "Test the complete and reduced SVDs."
+  (labels ((mul-diag-times-gen (diag matrix)
+             "Returns a newly allocated matrix resulting from the product of DIAG (a diagonal real matrix) with MATRIX (a complex matrix)."
+             (declare (type matrix diag matrix)
+                      (values matrix))
```

Comment:
`(values matrix)` ? what sorcery is this?

Reply:
I didn't realize that this is a CMUCL/SBCLism and is not in the ANSI standard. It's quite handy, though.

The `(declare (values type))` declaration is equivalent to wrapping the last form in a function with `(the type (my-fancy-form))` but cleaner.

Reply:
Neat. TIL. That was my assumption, and when I jumped over to your quilc PR, I noticed it there as well and that pretty much confirmed it.  Not sure how I never noticed this before while scrounging around in the SBCL sources!

Reply:
I don't think I understand.  It's equivalent, below, to doing `(matrix (dotimes (i (min ... `?  But that function doesn't return a matrix?

Reply:
Oh nvm.  I see that it does return `result`.  So you're declaring the return type of the function?

Reply:
@notmgsk That's right. However, since it is a sbcl-ism I'm not sure if I should keep using it in our code. I do like it a lot more than declaring `ftype` or annotating return forms with `the`.

Reply:
Is it an error on other implementations?  If not, I don't think keeping it is harmful.

Reply:
I just tested ECL 16.1.3, Clisp 2.49.93+, and CCL 1.11.5 and only the latter produces a compiler warning (while still compiling the function just fine).